test/default_style.jl

```diff
@@ -4836,4 +4844,119 @@ some_function(
                arg2"""
         @test fmt(str_, 4, 1) == str
     end
+
+    @testset "Format docstrings" begin
+        unformatted = """
+        \"""
+        This is a docstring
+
+        ```@example
+        a =  1
+         b  = 2
+         a + b
+        ```
+
+        ```jldoctest
+        a =  1
+        b  = 2
+        a + b
+
+        # output
+
+        3
+        ```
+
+        ```jldoctest
+        julia> a =  1
+        1
+
+        julia> b  = 2;
+
+        julia>  a + b
+        3
+
+        julia> function test(x)
+               x + 1
+               end;
+        ```
+        \"""
+        function test(x) x end
+        """
+
+        formatted = """
+        \"""
+        This is a docstring
+
+        ```@example
+        a = 1
+        b = 2
+        a + b
+        ```
+
+        ```jldoctest
+        a = 1
+        b = 2
+        a + b
+
+        # output
+
+        3
+        ```
+
+        ```jldoctest
+        julia> a = 1
+        1
+
+        julia> b = 2;
+
+
+        julia> a + b
+        3
+
+        julia> function test(x)
+                   x + 1
+               end;
+
+        ```
+        \"""
+        function test(x)
+            x
+        end"""
+        @test fmt(unformatted) == formatted
+    end
+    @testset "Multi-line indented code-blocks" begin
+        unformatted = """
+        \"""
+            format_text(
+            )
+        \"""
+        function format_text() end
+        """
+
+        formatted = """
+        \"""
+        ```
+        format_text(
+        )
+        ```
+        \"""
+        function format_text() end"""
+
+        @test format_text(unformatted) == formatted
```

Comment:
What's the rationale for reformatting like this?

Reply:
I don't know. `Markdown.plain` does it though